code/game/gamemodes/objective.dm

```diff
@@ -364,11 +408,11 @@
 	return 1
 
 
-/datum/objective/martyr
+/datum/objective/escape_obj/martyr

```

Comment:
So what if it kills as many people as it wants? That's for the ling to decide. I'm not bossing around it's playstyle here.

Also, I'm pretty sure the absorb objective has martyr_compatible set to 0 so it wouldn't combine.